src/acl.c

```diff
@@ -983,22 +1023,35 @@ int ACLSetUser(user *u, const char *op, ssize_t oplen) {
                 return C_ERR;
             }
 
-            unsigned long id = ACLGetCommandID(copy);
-            /* Add the subcommand to the list of valid ones, if the command is not set. */
-            if (ACLGetUserCommandBit(u,id) == 0) {
-                ACLAddAllowedSubcommand(u,id,sub);
+            if (cmd->subcommands_dict) {
+                /* If user is trying to allow a valid subcommand we can just add its unique ID */
+                struct redisCommand *cmd = ACLLookupCommand(op+1);
+                if (cmd == NULL) {
+                    errno = ENOENT;
+                    return C_ERR;
+                }
+                ACLChangeCommandPerm(u,cmd,1);
+            } else {
+                /* If user is trying to abuse the ACL mech to block SELECT 0 and alike we use the
```

Comment:
I don't think we wanna mention it's an abuse.. The original design (in Redis) wasn't very good and it allowed that. 
Also, since we didn't promote DEBUG to a container command, that's a perfect excuse for supporting this ("first argument" filtering)